### PR TITLE
feat(core): onetime subscription ui

### PR DESCRIPTION
**Pull request by CatsJuice**

### CODE REVIEW

**Comment by CatsJuice** [comment]
* **#8462** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8462?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#8369** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8369?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `canary`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @CatsJuice and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/frontend/core/src/components/affine/setting-modal/general-setting/plans/plan-card.tsx [hunk 1]
```diff
@@ -104,6 +112,7 @@ const ActionButton = ({ detail, recurring }: PlanCardProps) => {
   //    else                                    => 'Buy Pro'
   //  else
   //    if isBeliever                           => 'Included in Lifetime'
+  //    if onetime                              => 'Redeem Code'
   //    if isCurrent
   //      if canceled                           => 'Resume'
   //      else                                  => 'Current Plan'
```
Comment (forehalo) [inline packages/frontend/core/src/components/affine/setting-modal/general-setting/plans/plan-card.tsx:118]: <i class='graphite__hidden'>[Re: lines +108 to +122]<br/><br/></i>

this comment seems to be wrong<span class='graphite__hidden'><br/><br/>See this comment inline on <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8462?utm_source=unchanged-line-comment">Graphite</a>.</span>

Comment (CatsJuice) [reply packages/frontend/core/src/components/affine/setting-modal/general-setting/plans/plan-card.tsx:118]: Do you mean duplicated `isCurrent` judgement?

Comment (forehalo) [reply packages/frontend/core/src/components/affine/setting-modal/general-setting/plans/plan-card.tsx:118]: yup

**Review by forehalo — APPROVED** [review]

**Pull request merged**